### ISSUE TITLE
module error

### ISSUE THREAD

**Issue by pk372** [B1naryKill3r/HQSolver#36]
File "main.py" ,line 51 ,in lauchSolver
       contextFactory=ssl.ClientContextFactory()
AttributeError: module 'ssl' has no attribute 'ClientContextFactory'

Please reply ...And Suggest me What to do

**Comment by pk372** [comment]
i fixed this issue.but another issue occured which i won't able to remove please help

**Comment by Alphxx101** [comment]
Mine works perfectly fine. Maybe make sure that you have MSV Studio 14 at least and follow instructions properly. Delete and re install?

**Comment by Alphxx101** [comment]
If you run 
`python main.py`
Does it say this?
![image](https://user-images.githubusercontent.com/45533484/49403431-a9003500-f744-11e8-85d9-fbd529aa6396.png)

**Comment by pk372** [comment]
i have MSV 14,
it works but when Question comes it stops

**Comment by Alphxx101** [comment]
Oh well I don't know then. I only got it installed today after game. I'll try tomorrow in the game and see if mine works or stops.

**Comment by jirgon** [comment]
fwiw, i get the same error `AttributeError: module 'ssl' has no attribute 'ClientContextFactory'` when the show is live

**Comment by ActionLJackson** [comment]
AttributeError: module 'ssl' has no attribute 'ClientContextFactory'

**Comment by pk372** [comment]
I resolve that iisue

**Comment by pk372** [comment]
Only u have to do is in.  main.py

Delete import ssl and write:from twisted.internet import ssl
Simply say interchange the position of import ssl and from twisted.internet import ssl

**Comment by Alphxx101** [comment]
FYI, mine shows this right now. Does yours show this?
![image](https://user-images.githubusercontent.com/45533484/49469463-83843180-f7ff-11e8-82a8-8ce142b6c2e9.png)

**Comment by pk372** [comment]
yup

**Comment by pk372** [comment]
it display all things ,player counts e.t.c but when Question comes suddenly error comes

**Comment by Alphxx101** [comment]
Aha, this is for US HQ I guess. I am UK no wonder the show is live. Mine says the same error as you
![image](https://user-images.githubusercontent.com/45533484/49469885-8df2fb00-f800-11e8-9129-f851fe92fcfe.png)

**Comment by Alphxx101** [comment]
The lines are about the `self.question`, so i just removed those lines. I will try when game is live again

**Comment by pk372** [comment]
no ,tempPrint_=self.question["question"].encode( "UTF-8" ) returning byte data type.
it should be str, i didn't get this error first but after reinstalling python i get exactly same error.
Just add        replace the above line by    tempPrint+=str(self.question["question"].encode( "UTF-8" ))

**Comment by Alphxx101** [comment]
So that fixed it for you?

**Comment by Alphxx101** [comment]
UK HQ is live right now but this bot doesn't work with that?

**Comment by pk372** [comment]
mine working fine  in HQ US,

**Comment by Alphxx101** [comment]
Yeah mine works with HQ US but when the question shows up if it has speech marks " " the it shows this.

`b'Art lovers and \xe2\x80\x9cBlues Brothers\xe2\x80\x9d fans know that Chicago\xe2\x80\x99s Daley Plaza features a sculpture by who?'`

Also is the question supposed to start with "b" everytime?

**Comment by pk372** [comment]
giving 0 results

**Comment by pk372** [comment]
Trying to connect with api

**Comment by Alphxx101** [comment]
What do you mean.

**Comment by pk372** [comment]
no search is found

**Comment by Alphxx101** [comment]
Oh, does it not use Google Custom Search API? How would we connect it with that.

**Comment by pk372** [comment]
no ,it doesn't uses google custom search api,   there should be some changes in search.py .whenerver i got time i'll correct it

**Comment by Alphxx101** [comment]
OK for sure. You have discord?

**Comment by pk372** [comment]
hmmmm

**Comment by Alphxx101** [comment]
OK, add me if you want. It's easier to talk there. If not then here is still fine. Alphxx#9811

**Comment by Alphxx101** [comment]
Yeah, I just looked in search.py. Im guessing this is what you were talking about?
`async def search_google(question, num_results):
    """
    Returns num_results urls from a google search of question.
    :param question: Question to search
    :param num_results: Number of results to return
    :return: List of length num_results of urls retrieved from the search
    """
    # Could use Google's Custom Search API here, limit of 100 queries per day
    # result = service.cse().list(q=question, cx=CSE_ID, num=num_results).execute()
    # return result["items"]
    page = await networking.get_response(GOOGLE_URL.format(question), timeout=5, headers=HEADERS)
    return get_google_links(page, num_results)`

**Comment by Alphxx101** [comment]
async def search_google(question, num_results):
    """
    Returns num_results urls from a google search of question.
    :param question: Question to search
    :param num_results: Number of results to return
    :return: List of length num_results of urls retrieved from the search
    """
    # Could use Google's Custom Search API here, limit of 100 queries per day
    # result = service.cse().list(q=question, cx=CSE_ID, num=num_results).execute()
    # return result["items"]
    page = await networking.get_response(GOOGLE_URL.format(question), timeout=5, headers=HEADERS)
    return get_google_links(page, num_results)
